docs/integrations/r.md

```diff
@@ -0,0 +1,177 @@
+```
+
+In conjunction with this you must also specify `region` and `use_https` _in each call of an `aws.s3` function_ as these cannot be set globally. For example: 
+
+```r
+bucketlist(
+    region="",
+    use_https="false"
+    )
+```
+
+* `region` should always be empty
+* `use_https` should be set to `true` or `false` depending on whether your lakeFS endpoint uses HTTPS.
+
+### Listing repositories
+
+The S3 gateway exposes a repository as a bucket, and so using the `aws.s3` function `bucketlist` will return a list of available repositories on lakeFS: 
+
+```r
+bucketlist(
+    region="",
+    use_https="false"
+    )
+```
+
+### Writing to lakeFS from R
+
+Assuming you're using the `aws.s3` library there various functions available including `s3save`, `s3saveRDS`, and `put_object`. Here's an example of writing an R object to lakeFS: 
+
+```r
+repo_name <- "example"
+branch <- "development"
+
+s3saveRDS(x=my_df, 
+          bucket = repo_name, 
+          object = paste0(branch,"/my_df.R"), 
+          region="",
+          use_https="false")
+```
+
+You can also upload local files to lakeFS using R and the `put_object` function: 
+
+```r
+repo_name <- "example"
+branch <- "development"
+local_file <- "/tmp/never.gonna"
+
+put_object(file = local_file, 
+           bucket = repo_name, 
+           object = paste0(branch,"/give/you/up"),
+           region="",
+           use_https=useHTTPS)
+```
+
+### Reading from lakeFS with R
+
+As with writing data from R to lakeFS, there is a similar set of functions for reading data. These include `s3load`, `s3readRDS`, and `get_object`. Here's an example of reading an R object from lakeFS: 
+
+```r
+repo_name <- "example"
+branch <- "development"
+
+my_df <- s3readRDS(bucket = repo_name, 
+                   object = paste0(branch,"/my_data.R"),
+                   region="",
+                   use_https=useHTTPS)
+```
+
+## Performing lakeFS Operations using the lakeFS API from R
+
+As well as reading and writing data, you will also want to carry out lakeFS operations from R including creating branches, committing data, and more. 
+
+To do this call the lakeFS [API](https://docs.lakefs.io/reference/api.html) from the `httr` library. You should refer to the API documentation for full details of the endpoints and their behaviour. Below are a few examples to illustrate the usage. 
+
+### Check the lakeFS Server Version
+
+This is a useful API call to establish connectivity and test authentication. 
+
+```r
+library(httr)
+lakefs_api_url<- "lakefs.mycorp.com:8000/api/v1"
+lakefsAccessKey <- 'AKIAIOSFODNN7EXAMPLE'
+lakefsSecretKey <- 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+
+r=GET(url=paste0(lakefs_api_url,"/config/version"), 
+      authenticate(lakefsAccessKey, lakefsSecretKey))
+```
+
+The returned object `r` can be inspected to determine the outcome of the operation. Here is some example R code to demonstrate the idea: 
+
+```r
+if (r$status_code <400) {
```

Comment:
Suggest to guide the user to check the spec and in the version example compare with 200 it is the only status code for this call that should return successfully the version information.

Reply:
good point. done.